File: baekjoon/1021.js
function solution(n, m, positions) {
  // 1부터 n까지의 숫자로 배열 만들기
  let queue = Array.from({ length: n }, (_, i) => i + 1);
  let count = 0;

  for (let target of positions) {
    while (true) {
      // 목표가 맨 앞에 있으면 뽑기
      if (queue[0] === target) {
        queue.shift(); // 맨 앞 원소 제거
        break;
      }

      // 목표의 위치 찾기
      let targetIdx = queue.indexOf(target);

      // 왼쪽으로 가는게 더 빠른지 판단
      if (targetIdx <= queue.length / 2) {
        // 왼쪽으로 한 칸 이동
        queue.push(queue.shift());
        count++;
      } else {
        // 오른쪽으로 한 칸 이동
        queue.unshift(queue.pop());
        count++;
      }
    }
  }

  return count;
}

console.log(solution(10, 3, [1, 2, 3])); // 위치에 따른 최소 이동 횟수 출력

File: baekjoon/1021.py
from collections import deque  # 큐를 사용하기 위한 라이브러리

def solution(n, m, positions):
    queue = deque(range(1, n+1))  # 1부터 n까지의 숫자로 큐 초기화
    count = 0  # 2,3번 연산의 총 횟수
    
    for target in positions:  # 뽑아야 할 각 숫자에 대해
        while True:
            # 뽑으려는 수가 맨 앞에 있으면
            if queue[0] == target:
                queue.popleft()  # 뽑기
                break
                
            # 뽑으려는 수의 위치 찾기
            target_idx = queue.index(target)
            
            # 왼쪽으로 가는게 더 빠른지, 오른쪽으로 가는게 더 빠른지 판단
            if target_idx <= len(queue) // 2:  # 왼쪽이 더 가까우면
                queue.rotate(-1)  # 왼쪽으로 한 칸 이동
                count += 1
            else:  # 오른쪽이 더 가까우면
                queue.rotate(1)  # 오른쪽으로 한 칸 이동
                count += 1
                
    return count

n = 10  # 큐의 크기
m = 3   # 뽑을 숫자의 개수
positions = [1, 2, 3]  # 뽑아야 할 숫자들
print(solution(n, m, positions))